### PR TITLE
Refactor CSI RBD plugin to V0.2.0

### PR DESCRIPTION
**Pull request by sbezverk**
This PR does:
- Refactor to CSI v0.2.0 spec
- Adds required by CSI spec checks to Controller server
- Introduces rbdVolumes map which:
      loads at init time all old volumes stored in controller folder
      keeps all known volumes
      allows to implement in future ListVolumes Controller call

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -1,23 +1,31 @@
 [[constraint]]
-  branch = "master"
   name = "github.com/container-storage-interface/spec"
+  version = "~0.2.0"
```
Comment (rootfs) [inline Gopkg.toml]: ~0.2.0?

File: Dockerfile [hunk 2]
```diff
@@ -7,6 +7,6 @@ RUN yum  install -y centos-release-ceph && \
     yum install -y ceph-common e2fsprogs && \ 
     yum clean all
 
-COPY rbdplugin /rbdplugin
+COPY _output/rbdplugin /rbdplugin
```
Comment (rootfs) [inline Dockerfile]: add `/rbdplugin` to gitignore

File: pkg/rbd/controllerserver.go [hunk 3]
```diff
@@ -128,11 +154,3 @@ func (cs *controllerServer) ValidateVolumeCapabilities(ctx context.Context, req
 	}
 	return &csi.ValidateVolumeCapabilitiesResponse{Supported: true, Message: ""}, nil
 }
-
-func (cs *controllerServer) ControllerUnpublishVolume(ctx context.Context, req *csi.ControllerUnpublishVolumeRequest) (*csi.ControllerUnpublishVolumeResponse, error) {
```
Comment (rootfs) [inline pkg/rbd/controllerserver.go]: keep this, we'll need it to do rbd fencing

File: pkg/rbd/rbd.go [hunk 4]
```diff
@@ -73,16 +124,15 @@ func NewNodeServer(d *csicommon.CSIDriver) *nodeServer {
...
 	// Initialize default library driver
-	rbd.driver = csicommon.NewCSIDriver(driverName, &version, GetSupportedVersions(), nodeID)
+	rbd.driver = csicommon.NewCSIDriver(driverName, version, nodeID)
 	if rbd.driver == nil {
 		glog.Fatalln("Failed to initialize CSI Driver.")
 	}
 	rbd.driver.AddControllerServiceCapabilities([]csi.ControllerServiceCapability_RPC_Type{
 		csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME,
-		csi.ControllerServiceCapability_RPC_PUBLISH_UNPUBLISH_VOLUME,
 	})
```
Comment (rootfs) [inline pkg/rbd/rbd.go]: keep this

File: Gopkg.toml (hunk 1 above)
Comment (sbezverk) [reply Gopkg.toml]: It is done the same way across all CSI external components (registart, attacher, provisioner, etc)

File: rbd/main.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ func init() {
 
 var (
 	endpoint   = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
-	driverName = flag.String("drivername", "rbdplugin", "name of the driver")
+	driverName = flag.String("drivername", "csi-rbdplugin", "name of the driver")
```
Comment (rootfs) [inline rbd/main.go]: update README and deployment too

File: Dockerfile (hunk 2 above)
Comment (sbezverk) [reply Dockerfile]: done

File: pkg/rbd/controllerserver.go (hunk 3 above)
Comment (sbezverk) [reply pkg/rbd/controllerserver.go]: done

File: pkg/rbd/rbd.go (hunk 4 above)
Comment (sbezverk) [reply pkg/rbd/rbd.go]: done

**Comment by rootfs** [comment]
let's have this! thanks

File: rbd/main.go (hunk 5 above)
Comment (sbezverk) [reply rbd/main.go]: done

**Comment by sbezverk** [comment]
@rootfs addressed your comments, PTAL

File: deploy/kubernetes/rbdplugin.yaml [hunk 6]
```diff
@@ -47,35 +44,35 @@ roleRef:
...
             - name: KUBE_NODE_NAME
               valueFrom:
                 fieldRef:
                   fieldPath: spec.nodeName
           volumeMounts:
             - name: socket-dir
-              mountPath: /var/lib/kubelet/plugins/rbdplugin
-        - name: rbdplugin 
+              mountPath: /var/lib/kubelet/plugins/csi-rbdplugin
+        - name: cai-rbdplugin
```
Comment (rootfs) [inline deploy/kubernetes/rbdplugin.yaml]: csi

**Pull request merged**